packages/wrangler/src/api/upgrade.ts

```diff
@@ -0,0 +1,94 @@
+import { writeFileSync, existsSync } from "fs";
+import path from "path";
+import { execaSync } from "execa";
+import { fetch } from "undici";
+import { findWranglerToml } from "../config";
+import { confirm } from "../dialogs";
+import { printWranglerBanner } from "../index";
+import { logger } from "../logger";
+import { readFileSync } from "../parse";
+
+import type { PackageJSON } from "../parse";
+
+export async function upgradeWrangler({
+	yesFlag = false,
+}: {
+	yesFlag: boolean;
+}) {
+	await printWranglerBanner();
+	let packageJsonData: undefined | PackageJSON;
+
+	const pathToWorker = findWranglerToml();
+	if (pathToWorker) {
+		const newPath = pathToWorker.split(path.sep).slice(0, -1).join(path.sep);
```

Comment:
This feels very brittle—what about something like `path.resolve(pathToWorker, "./package.json")`?

Reply:
What do you mean by brittle?

Reply:
Maybe `path.dirname()`? It just seems that explicitly splitting path and mutating them like this is less clear than using the APIs in the path module. Effectively, this code is doing `path.join(path.dirname(findWranglerToml()), "package.json")`, but it feels (to me, anyway), that that's a lot better for future readability than string manipulation of paths.